media-libs/mesa/mesa-9999.ebuild

```diff
@@ -183,7 +183,10 @@ DEPEND="${RDEPEND}
 BDEPEND="
 	${PYTHON_DEPS}
 	opencl? (
-		>=sys-devel/gcc-4.6
+		>=virtual/rust-1.62.0
+		>=dev-util/bindgen-0.58.0
+		dev-util/spirv-tools
+		dev-util/spirv-llvm-translator
```

Comment:
The [Rusticl documentation](https://docs.mesa3d.org/rusticl.html) says the build dependencies are

>- rustc
>- rustfmt (highly recommended, but only required for CI builds or when authoring patches)
>- bindgen
>- [LLVM](https://github.com/llvm/llvm-project/) built with libclc and -DLLVM_ENABLE_DUMP=ON.
>- [SPIRV-Tools](https://github.com/KhronosGroup/SPIRV-Tools)
>- [SPIRV-LLVM-Translator](https://github.com/KhronosGroup/SPIRV-LLVM-Translator) for a libLLVMSPIRVLib.so matching your version of LLVM, i.e. if you’re using LLVM 15 (libLLVM.so.15), then you need a libLLVMSPIRVLib.so.15.

The dependency on `dev-util/spirv-llvm-translator` is certainly wrong:

````
/var/tmp/portage/media-libs/mesa-9999/image # scanelf -n ./usr/lib64/libRusticlOpenCL.so
 TYPE   NEEDED FILE 
ET_DYN libLLVM-15.so,libz.so.1,libzstd.so.1,libdrm.so.2,libexpat.so.1,libclang-cpp.so.g1,libLLVMSPIRVLib.so.15,libSPIRV-Tools-opt.so,libSPIRV-Tools.so,libSPIRV-Tools-link.so,libstdc++.so.6,libm.so.6,libgcc_s.so.1,libc.so.6,ld-linux-x86-64.so.2 ./usr/lib64/libRusticlOpenCL.so 
````

It should be in `DEPEND` and `RDEPEND`. I believe it should **not** be in `BDEPEND`, since I don't see any invocations of `llvm-spirv` in the `build.log`:

````
/var/tmp/portage/media-libs/mesa-9999 # grep llvm-spirv temp/build.log
<empty>

````

Additionally, it needs to be linked against the same version of LLVM otherwise used by Mesa.

>The minimum versions to build Rusticl are:
>
> - Rust: 1.59

I don't know why you chose `>=virtual/rust-1.62.0`, but that's fine with me. 1.66 is stable.

> - Meson: 1.0.0

`>=dev-util/meson-1.0.0` definitely needs to be specified as a dependency in `BDEPEND`. `meson.eclass` only requires 0.62.2.

> - Bindgen: 0.58.0

I tested with bindgen-0.56.0 and it failed. bindgen-0.58 isn't in ::gentoo, so I assume this is fine.

> - LLVM: 11.0.0 (recommended: 15.0.0)

We only allow `sys-devel/llvm:15` or `sys-devel/llvm:16` in `mesa-*.ebuild`, so this is good.

> - SPIRV-Tools: any version (recommended: v2022.3)

We package the Vulkan-SDK tags and not the date-based tags, so I don't know how this compares exactly. `dev-util/spirv-tools-1.3.243`'s `/usr/lib64/pkgconfig/SPIRV-Tools.pc` says `Version: 2023.2.1` so it's definitely new enough. I suggest simply requiring `>=dev-util/spirv-tools-1.3.243` and not worrying about the precise lower-bound.

But, like `dev-util/spirv-llvm-translator` this should be in `DEPEND` and `RDEPEND` (and probably **not** in `BDEPEND` because I see no invocations of `spirv-*` programs in the `build.log`). Note that in the `scanelf` output above that Rusticl is linking with `libSPIRV-Tools-opt.so,libSPIRV-Tools.so,libSPIRV-Tools-link.so`.







 I will handle these updates. Thank you for your contribution, and I hope that you'll contribute in the future.

At the same time, I would like to take this opportunity say something to those that thought that all that was needed from my side was to press a button:

- It was not.
- It almost never is.
- Please be patient.
- Your pings do not help.
- They just add stress to a volunteer who is already doing much more than he wants to or feels is sustainable.
- If you want to see something happen sooner, please try to take something else off my plate.

Reply:
> I don't know why you chose >=virtual/rust-1.62.0

I honestly don't remember. Probably because that was what I initially tested it against and since it was already stable it didn't matter that much.

> >=dev-util/meson-1.0.0 definitely needs to be specified as a dependency

It's one of those deps that got upgraded recently because initially it required a much older version.

> I will handle these updates

Thanks!

> Thank you for your contribution, and I hope that you'll contribute in the future.

Thank you for your review, it's much appreciated!

> At the same time, I would like to take this opportunity say something to those that thought that all that was needed from my side was to press a button

I don't think that anybody thought that a press of button was all that was needed and the long wave of iterations this PR has gone through is the proof that things are never that simple. I personally just wanted to reassure you that I'm actually using this thing and I won't disappear once the PR gets merged.

> They just add stress to a volunteer who is already doing much more than he wants to or feels is sustainable

I know the feeling, I'm currently struggling to succesfully compile latest Chromium 113 on ppc while I'm bisecting the kernel because 6.3 decided to read random memory pages and fails to boot, latest vscode stopped working against my custom ppc64 electron build and qtwebengine regressed as well. All while trying to get some actual work done, which heavily relies on having working Chromium/electron ebuilds since I'm a full stack Javascript developer.